### PR TITLE
Fixes Issue #638 - Address unreliability of test_opening_every_category_page_from_categories_menu

### PR DESCRIPTION
**Pull request by bobsilverberg**
Improve performance by caching elements
Only check the first 3 categories and apps

I confirmed that the second change is ok with @krupa.

Note: when reviewing this it might be worthwhile running the test in its current form, to observe the performance issues, and then run the new version.

@rbillings / @m8ttyB r?

### CODE REVIEW

File: tests/desktop/consumer_pages/test_home_page.py [hunk 1]
```diff
@@ -52,26 +52,31 @@ def test_that_verifies_categories_menu(self, mozwebqa):
...
-    def test_opening_every_category_page_from_categories_menu(self, mozwebqa):
+    def test_opening_category_pages_from_categories_menu(self, mozwebqa):
+        """Open the first 3 category pages and check the first 3 apps on those pages."""
 
         home_page = Home(mozwebqa)
         home_page.go_to_homepage()
 
-        for i in range(home_page.category_count):
+        categories = home_page.categories.items
+        for c in range(3):
```
Comment (m8ttyB) [inline tests/desktop/consumer_pages/test_home_page.py]: nit, add a code comment explaining the logic for `range(3)`

File: tests/desktop/consumer_pages/test_home_page.py [hunk 2]
```diff
@@ -52,26 +52,31 @@ def test_that_verifies_categories_menu(self, mozwebqa):
...
+            Assert.true(len(apps) > 0)
             Assert.true(category_page.is_new_popular_tabs_visible)
             Assert.true(category_page.popular_tab_class == 'active')
 
-            for i in range(len(category_page.apps)):
-                Assert.true(category_page.apps[i].is_name_visible)
-                Assert.true(category_page.apps[i].is_icon_visible)
-                Assert.true(category_page.apps[i].is_rating_visible)
-                Assert.true(category_page.apps[i].is_install_visible)
+            for a in range(3):
```
Comment (m8ttyB) [inline tests/desktop/consumer_pages/test_home_page.py]: Same nit as above, add a code comment

File: tests/desktop/consumer_pages/test_home_page.py (hunk 1 above)
Comment (bobsilverberg) [inline tests/desktop/consumer_pages/test_home_page.py]: I added a docstring at the top of the test that explains that we are checking the first 3 categories and the first 3 apps on the page, so I thought that might be sufficient. I can also add one here if you think that's needed.

**Comment by bobsilverberg** [comment]
@m8ttyB I added the comments. r?

File: tests/desktop/consumer_pages/test_home_page.py [hunk 3]
```diff
@@ -52,26 +52,31 @@ def test_that_verifies_categories_menu(self, mozwebqa):
...
-            category_name = home_page.categories.items[i].name
-            category_page = home_page.categories.items[i].click_category()
+            category = categories[c]
+            category_name = category.name
+            category_page = category.click_category()
             Assert.equal(category_name.title(), category_page.category_title)
             Assert.true(category_page.is_the_current_page)
-            Assert.true(len(category_page.apps) > 0)
+            apps = category_page.apps
+            Assert.true(len(apps) > 0)
```
Comment (m8ttyB) [inline tests/desktop/consumer_pages/test_home_page.py]: Just to verify, @krupa is OK with doing a less verbose verification, only verifying the apps `count is greater than 0`?

**Comment by m8ttyB** [comment]
r+

**Comment by bobsilverberg** [comment]
Thanks @m8ttyB!

**Pull request merged**